src/Nethermind/Nethermind.Synchronization/Blocks/BlockDownloader.cs

```diff
@@ -341,6 +341,10 @@ bool HasMoreToSync()
 
                     Block currentBlock = blocks[blockIndex];
                     if (_logger.IsTrace) _logger.Trace($"Received {currentBlock} from {bestPeer}");
+                    if (currentBlock.Header.HasBody && currentBlock.Body.IsEmpty)
+                    {
```

Comment:
why changes needed?

Reply:
Explained on prev comment. Before, when any issue happen during block download, then it will throw and not attempt to process at all. This will allow it to process part of the block body.

Reply:
Explained on prev comment. Before, when any issue happen during block download, then it will throw and not attempt to process at all. This will allow it to process part of the block body.